src/distPlus/distribution/XYShape.re

```diff
@@ -324,8 +378,31 @@ module Range = {
 
   let derivative = mapYsBasedOnRanges(delta_y_over_delta_x);
 
-  // TODO: It would be nicer if this the diff didn't change the first element, and also maybe if there were a more elegant way of doing this.
+  let stepwiseToLinear = ({xs, ys}: T.t): T.t => {
+    // adds points at the bottom of each step.
+    let length = E.A.length(xs);
+    let newXs: array(float) = Belt.Array.makeUninitializedUnsafe(2 * length);
+    let newYs: array(float) = Belt.Array.makeUninitializedUnsafe(2 * length);
+
+    let _ = Belt.Array.set(newXs, 0, xs[0] -. epsilon_float);
+    let _ = Belt.Array.set(newYs, 0, 0.);
+    let _ = Belt.Array.set(newXs, 1, xs[0]);
+    let _ = Belt.Array.set(newYs, 1, ys[0]);
```

Comment:
I may have been guilty of this before, but I think the way we are supposed to deal with this issue is by doing,
```Belt.Array.set(newXs, 1, xs[0]) |> ignore ```
https://github.com/reasonml-community/error-message-improvement/issues/33

Reply:
Nice! Thanks